src/algorithms.jl

```diff
@@ -0,0 +1,84 @@
+abstract type AbstractEnclosureAlgorithm end
+
+"""
+   NaturalEnclosure <: AbstractEnclosureAlgorithm
+
+Data type to compute the range of `f` over `X` using natural enclosure, that is simply
+evaluates `f(X)` using interval arithmetic.
+"""
+struct NaturalEnclosure <: AbstractEnclosureAlgorithm end
+
+"""
+    MooreSkelboeEnclosure{T} <: AbstractEnclosureAlgorithm
+
+Data type to compute the range of `f` over `X` using the Moore-Skelboe algorithm, which
+computes rigorously the global minimum and maximum of the function.
+See [`IntervalOptimisation.jl`](https://github.com/JuliaIntervals/IntervalOptimisation.jl) for more details.
+
+### Fields
+
+- `structure` -- (default: `HeapedVector`) the way in which vector elements
+                 are kept arranged; possible options are `HeapedVector` and `SortedVector`
+- `tol`       -- (default `1e-3`) tolerance to which the optima are computed
+
+### Notes
+
+To use this algorithm, you need to import `IntervalOptimisation.jl`
+"""
+struct MooreSkelboeEnclosure{T} <: AbstractEnclosureAlgorithm
```

Comment:
Here and in the rest of structs containing fields I suggest that we use `Base.@kwdef` and remove the outer constructor.